Sources/Gravatar/Network/Services/ImageUploadService.swift

```
@@ -18,6 +18,11 @@ struct ImageUploadService: ImageUploader {
         avatarSelection: AvatarSelection = .preserveSelection,
         additionalHTTPHeaders: [HTTPHeaderField]?
     ) async throws -> (data: Data, response: HTTPURLResponse) {
+        assert(
+            image.isSquare(),
+            "This image is not square, and uploading it will fail. Consider using `SquaringStrategy.default` when calling `AvatarService.upload(::::)`."
+        )
```

Is this a good idea?  It would help developers discover non-square images in development (DEBUG).  But will it interfere with development somehow?

Should we drop this in favor of logging something to Console?

---

Maybe is better to return the error from the server in this case. 
We don't know if/when the backend will start allowing non-squared images, so it's probably better to let the image reach the backend and it can decide? 🤔

---

Now that we have a clear error message coming from the backend, i think it is fine to defer this to backend and remove this assert.